blockchain/types/transaction.go

```diff
@@ -1024,16 +1024,21 @@ func (t *TransactionsByTimeAndNonce) Pop() {
 }
 
 // NewMessage returns a `*Transaction` object with the given arguments.
-func NewMessage(from common.Address, to *common.Address, nonce uint64, amount *big.Int, gasLimit uint64, gasPrice *big.Int, data []byte, checkNonce bool, intrinsicGas uint64) *Transaction {
+func NewMessage(from common.Address, to *common.Address, nonce uint64, amount *big.Int, gasLimit uint64, gasPrice *big.Int, data []byte, checkNonce bool, intrinsicGas uint64, list AccessList) *Transaction {
 	transaction := &Transaction{
 		validatedIntrinsicGas: intrinsicGas,
 		validatedFeePayer:     from,
 		validatedSender:       from,
 		checkNonce:            checkNonce,
 	}
 
-	internalData := newTxInternalDataLegacyWithValues(nonce, to, amount, gasLimit, gasPrice, data)
-	transaction.setDecoded(internalData, 0)
+	if list != nil {
```

Comment:
Hardfork condition seems required. What do you think?

Reply:
I've looked into references to `NewMessage`, and it seems they are all related to calls (e.g., `klay_call` API).
Since calls do not affect states, hardfork condition could be omitted.

Reply:
Could you check it out again? I found one:

Call, EstimateGas (api_ethereum.go) -> EthDoCall (api_ethereum.go) -> args.ToMessage (api_ethereum.go) -> ToMessage (tx_args.go) -> NewMessage (transaction.go)

Reply:
They are 'call's, right? The `eth_call` or `eth_estimateGas` could behave differently, however that won't change the states.
Could you help me with a concrete example for problematic case? I cannot come up with one.

Reply:
Aha, yes, it would not be required. I'm sorry I overlooked your first comment. Thank you.